icu4c/source/common/dictbe.cpp

```diff
@@ -1077,14 +1078,22 @@ CjkBreakEngine::CjkBreakEngine(DictionaryMatcher *adoptDictionary, LanguageType
         UnicodeSet cjSet(UnicodeString(u"[[:Han:][:Hiragana:][:Katakana:]\\u30fc\\uff70\\uff9e\\uff9f]"), status);
         if (U_SUCCESS(status)) {
             setCharacters(cjSet);
+
+#if !UCONFIG_NO_CJ_ML_PHRASE_BREAKING
+            fMlBreakEngine = new CjMLBreakEngine(fDigitOrOpenPunctuationOrAlphabetSet, fClosePunctuationSet, status);
+#else
             initJapanesePhraseParameter(status);
+#endif
         }
     }
     UTRACE_EXIT_STATUS(status);
 }
 
 CjkBreakEngine::~CjkBreakEngine(){
     delete fDictionary;
+    if (fMlBreakEngine != nullptr) {
+        delete fMlBreakEngine;
```

Comment:
just call "delete fMlBreakEngine;" . delete will handle nullptr. 
see https://www.cpp-junkie.com/2020/11/c-is-it-safe-to-delete-null-pointer.html

Reply:
Done